content/blog/2019/05/2019-05-30-becoming-contributor-newbie-tickets.adoc

```diff
@@ -0,0 +1,243 @@
+Maven projects. Note that instructions to install JDK and Maven are linked in the contributing guidelines.
+
+Once we have all the necessary tools installed and configured, we are ready to https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#building-and-debugging[build Jenkins locally]
+and also to https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#testing-changes[run tests].
+
+== Getting down to business
+
+=== Reviewing ticket details
+
+Now that I was ready to start working on the ticket, I had to review it in more detail, to fully understand the problem.
+
+The description of the ticket I was planning to work on included two links. The first one was to
+https://github.com/jenkinsci/jenkins/pull/3715#issuecomment-470873111[a screenshot that showed the actual bug]. It showed
+how several non-compatible plugins were being selected when clicking “_All_”, even though the intended behavior was to
+only select the compatible plugins. The second link was https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/hudson/PluginManager/table.jelly#L108...L120[a reference to a code fragment]
+that showed other validations that had to be taken into account when checking if a plugin update was compatible or not
+with the current installation.
+
+=== Reproducing the issue locally
+
+Even though I had now understood the issue in better detail, I had not seen it myself live yet, so I seemed to me that
+the next logical step was to reproduce it locally.
+
+To reproduce the issue locally in our computer, we can either use the local `war` file that we can generate by
+https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md#building-and-debugging[building Jenkins from the source code]
+or we can download the https://jenkins.io/download/[latest Jenkins version] available and run it locally. When I worked
+on this ticket, the latest available version was 2.172` and, when I built if from the sources I saw version `2.173-SNAPSHOT`,
+which was the next version, in which the community was already working on.
+
+In general it is a good idea to reproduce bugs locally, not only to get a better understanding, but also to make sure
+they are actual issues. It could always be an issue happening only on the reporter’s end (e.g. some user misconfiguration).
+Or it could be a ticket referencing an old issue that has been fixed already. This latest possibility didn’t sound that
+strange to me, since the ticket was one month old. It could have been handled by someone else in the meantime, without
+noticing the ticket existed. Or the contributor might have forgotten to update the ticket in the issue tracker after the
+fix was committed.
+
+So, for all the reasons above, I went ahead in ran the latest Jenkins version locally. From a terminal, went to the
+folder in which the war file was placed, and ran `java -jar jenkins.war`, which starts Jenkins locally on http://localhost:8080.
```

Comment:
If you don't want the user to have a clickable link to http://localhost:8080/, then you could consider:

```suggestion
folder in which the war file was placed, and ran `java -jar jenkins.war`, which starts Jenkins locally on \http://localhost:8080.
```

Optional, since you may have intentionally made it a hyperlink for them to click.

Reply:
Thanks for the suggestion, Mark. In this particular case I was indeed having a link intentionally, to foster curiosity in the reader's end about running Jenkins locally :).